src/Microsoft.Azure.Relay/WebSockets/NetCore/WebSocketHandle.Managed.cs

```diff
@@ -99,9 +99,9 @@ public void Abort()
             {
                 Socket connectedSocket;
                 Stream stream;
-                if (options.Proxy != null && options.Proxy != WebRequest.DefaultWebProxy)
+                Uri proxyUri = options.Proxy != null && options.Proxy != WebRequest.DefaultWebProxy ? options.Proxy.GetProxy(uri) : null;
+                if (proxyUri != null)
```

Comment:
In the past I think I've seen cases where `GetProxy(uri)` returns the `uri` (maybe when a given address is configured to bypass proxy?).  Does that need to be handled?

Reply:
I guess this will just establish a direct connection, shouldn't be a problem.